spring/src/main/java/io/seata/spring/annotation/datasource/SeataDataSourceBeanPostProcessor.java

```diff
@@ -0,0 +1,108 @@
+/*
+ *  Copyright 1999-2019 Seata.io Group.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.seata.spring.annotation.datasource;
+
+import javax.sql.DataSource;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+
+import io.seata.common.exception.ShouldNeverHappenException;
+import io.seata.config.ConfigurationFactory;
+import io.seata.rm.datasource.DataSourceProxy;
+import io.seata.spring.util.SpringProxyUtils;
+import net.sf.cglib.proxy.Enhancer;
+import net.sf.cglib.proxy.MethodInterceptor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeanUtils;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+
+import static io.seata.core.constants.ConfigurationKeys.DATASOURCE_AUTOPROXY;
+import static io.seata.core.constants.ConfigurationKeys.DATASOURCE_USE_JDK_PROXY;
+
+/**
+ * @author xingfudeshi@gmail.com
+ * @date 2019/12/26
+ * The type seata data source bean post processor
+ */
+public class SeataDataSourceBeanPostProcessor implements BeanPostProcessor {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SeataDataSourceBeanPostProcessor.class);
+
+    @Override
+    public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
+        if (bean instanceof DataSource && !(bean instanceof DataSourceProxy) && ConfigurationFactory.getInstance().getBoolean(DATASOURCE_AUTOPROXY, false)) {
```

Comment:
Use EnableAutoDataSourceProxy, is ConfigurationKeys.DATASOURCE_AUTOPROXY still necessary?

Reply:
Probably no longer required.